url/src/lib.rs

```diff
@@ -3043,3 +3046,89 @@ impl<'a> Drop for UrlQuery<'a> {
         }
     }
 }
+
+#[cfg(all(feature = "uniffi", test))]
+uniffi::setup_scaffolding!();
+
+#[cfg(feature = "uniffi")]
+mod uniffi_core_impl {
+    use std::str::FromStr;
+
+    use super::*;
+
+    use uniffi_core::{
+        deps::bytes::{Buf, BufMut},
+        metadata, FfiConverter, MetadataBuffer,
+    };
+
+    uniffi_core::derive_ffi_traits!(blanket Url);
+
+    unsafe impl<UT> FfiConverter<UT> for Url {
+        uniffi_core::ffi_converter_rust_buffer_lift_and_lower!(UT);
```

Comment:
This `impl` "piggybacks" on UniFFI Core's impl of String, see https://github.com/mozilla/uniffi-rs/blob/main/uniffi_core/src/ffi_converter_impls.rs#L148-L169

For sake of simplicity, but this requires us to roundtrip via `to_string` on Url and via `Url::from_str`, there might be faster implementations, but this was the most convenient to get something started.